Sources/TensorFlow/Operators/NN.swift

```diff
@@ -90,6 +90,8 @@ public func conv2D<Scalar: TensorFlowFloatingPoint>(
     padding: Padding = .valid,
     dilations: (Int, Int, Int, Int) = (1, 1, 1, 1)
 ) -> Tensor<Scalar> {
+    precondition(input.shape.rank == 4, "input must have rank 4")
```

Comment:
Precondition and assertion messages should be a full sentence ([example](https://github.com/tensorflow/swift-apis/blob/6309c55e3acd71caa550c64b9d35c6c43fcaafbb/Sources/TensorFlow/Core/ShapedArray.swift#L607)) ending with a period.
```suggestion
    precondition(input.shape.rank == 4, "The input must have rank 4.")
```